### DISCUSSION TITLE
#9 📑 Weekly Report 9

### DISCUSSION THREAD

**Discussion by ghost** [SE-TINF22B6/Statify#73] [Blog]
**Current Status** 
Currently, our project focuses on two main areas. Firstly, we're actively developing a robust Streaming Daemon for the backend, essential for handling multiple streams concurrently and ensuring seamless user experiences. Secondly, we're planning and designing the database to store crucial user data and stream metadata. Our immediate priority is to finalize backend development and establish smooth integration with the frontend. Effective project management, including regular updates to our Kanban board, keeps us on track towards our goals of delivering a high-quality streaming application that meets user expectations.

**Last completed features**
We have updated the sequence diagrams and they will serve as invaluable visual aids, depicting the flow of interactions and communication between different components of our system. 

Furthermore, we have seamlessly integrated the risk management table into our project documentation. This table provides a comprehensive overview of potential risks and their corresponding mitigation strategies, enabling us to proactively identify and address potential challenges that may arise during the course of the project. 

**Plans for this semester and future essential features**
Our future plans entail implementing the streaming daemon and the database manager, followed by thorough testing of these features alongside the entire backend using dummy data. 
Additionally, we will implement the visual GUI for the web application, focusing on displaying the streaming data statistics of users prominently. Finally, we will ensure seamless integration between the backend and frontend components, ensuring smooth functionality and optimal user experience across the application.

**Architecture**
The current architecture lays the groundwork for implementing our tasks across various layers. At present, our plan involves utilizing a controller to manage communication between the frontend and backend. User-login information will reside in a Firebase database, while streaming statistics will be stored in a database yet to be determined. We will leverage Spring Boot, a powerful framework for building Java applications, to facilitate data retrieval. Spring Boot simplifies the development process by providing preconfigured setups and handling much of the boilerplate code, allowing us to focus on implementing business logic and features specific to our project. Additionally, authentication tokens will be employed to update streaming data from Spotify's API, ensuring that our application remains seamlessly integrated with external services.

**Comment by BenjaminB109** [comment]
Dear Statify-Team,
I look forward to seeing the GUI in an upcoming blog. Keep up the good work.

**Comment by mirak55** [comment]
Dear Statify team,
Your dedication to backend development and database design is awesome for laying a solid foundation for good user experience. Nice to also see some progress on sequence diagrams and your risk management table.
Keep up the work.

**Comment by Elizabethstella** [comment]
Nice progress! The focus on backend development and risk management integration is impressive. Looking forward to seeing the essential features come to life. Keep up the good work!
